aws-lc-rs/src/pqdsa/key_pair.rs

```diff
@@ -0,0 +1,372 @@
+    evp_pkey: LcPtr<EVP_PKEY>,
+    pubkey: PublicKey,
+}
+
+#[allow(clippy::missing_fields_in_debug)]
+impl Debug for PqdsaKeyPair {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PqdsaKeyPair")
+            .field("algorithm", &self.algorithm)
+            .finish()
+    }
+}
+
+impl KeyPair for PqdsaKeyPair {
+    type PublicKey = PublicKey;
+
+    fn public_key(&self) -> &Self::PublicKey {
+        &self.pubkey
+    }
+}
+
+/// A PQDSA seed.
+pub struct PqdsaSeed<'a>(&'a PqdsaKeyPair);
+
+// impl AsRawBytes<PqdsaSeedRaw<'static>> for PqdsaSeed<'_> {
+//     fn as_raw_bytes(&self) -> Result<PqdsaSeedRaw<'static>, Unspecified> {
+//         Ok(PqdsaSeedRaw::new(
+//             // Pending: https://github.com/aws/aws-lc/pull/2194
+//             todo!(),
+//         ))
+//     }
+// }
+
+/// A PQDSA private key.
+pub struct PqdsaPrivateKey<'a>(pub(crate) &'a PqdsaKeyPair);
+
+impl AsDer<Pkcs8V1Der<'static>> for PqdsaPrivateKey<'_> {
+    /// Serializes the key to PKCS#8 v1 DER.
+    ///
+    /// # Errors
+    /// Returns `Unspecified` if serialization fails.
+    fn as_der(&self) -> Result<Pkcs8V1Der<'static>, Unspecified> {
+        Ok(Pkcs8V1Der::new(
+            self.0
+                .evp_pkey
+                .marshal_rfc5208_private_key(pkcs8::Version::V1)?,
+        ))
+    }
+}
+
+impl AsRawBytes<PqdsaPrivateKeyRaw<'static>> for PqdsaPrivateKey<'_> {
+    fn as_raw_bytes(&self) -> Result<PqdsaPrivateKeyRaw<'static>, Unspecified> {
+        Ok(PqdsaPrivateKeyRaw::new(
+            self.0.evp_pkey.marshal_raw_private_key()?,
+        ))
+    }
+}
+
+impl PqdsaKeyPair {
+    /// Generates a new PQDSA key pair for the specified algorithm.
+    ///
+    /// # Errors
+    /// Returns `Unspecified` is the key generation fails.
+    pub fn generate(algorithm: &'static PqdsaSigningAlgorithm) -> Result<Self, Unspecified> {
+        let evp_pkey = evp_key_pqdsa_generate(algorithm.0.id.nid())?;
+        let pubkey = PublicKey::from_private_evp_pkey(&evp_pkey)?;
+        Ok(Self {
+            algorithm,
+            evp_pkey,
+            pubkey,
+        })
+    }
+
+    /// Parses a PKCS#8 key from the specified bytes.
+    ///
+    /// # Errors
+    /// Returns `Unspecified` if the key is not valid for the specified signing algorithm.
+    pub fn from_pkcs8(
+        algorithm: &'static PqdsaSigningAlgorithm,
+        pkcs8: &[u8],
+    ) -> Result<Self, Unspecified> {
+        let evp_pkey = LcPtr::<EVP_PKEY>::parse_rfc5208_private_key(pkcs8, EVP_PKEY_PQDSA)?;
+        validate_pqdsa_evp_key(&evp_pkey, algorithm.0.id)?;
+        let pubkey = PublicKey::from_private_evp_pkey(&evp_pkey)?;
+        Ok(Self {
+            algorithm,
+            evp_pkey,
+            pubkey,
+        })
+    }
+
+    /// Constructs a key pair from a raw private key.
+    ///
+    /// # Errors
+    /// Returns `Unspecified` if the key is not valid for the specified signing algorithm.
+    pub fn from_raw_private_key(
```

Comment:
I'm confused. What's the purpose of having both `from_raw_private_key` and `from_raw_seed`? Aren't they both the same? They both are just calling the same `EVP_PKEY_pqdsa_new_raw_private_key`. The documentation on seed says that the seeds are the same length, but then `parse_raw_private_key` implementation has an alternative length there. Is one of these functions implemented incorrectly?

Reply:
`from_raw_seed` is removed in the latest commit.